Slow Position Mode

MaEtUgR opened the pull request:
### Solved Problem
@marcinolokk and I implemented a slow version of the Position mode. It allows to fly like in Position mode but with slower horizontal and vertical velocity plus yaw turn rate. The sources for how much slower it goes in these dimensions can be:
- Parameters for metric limits as defaults
- Mappable auxiliary remote inputs e.g. knobs that allow scaling down
- uORB/MAVLink message specifying metric limits

Fixes #19799 (note that the default horizontal speed is 3m/s which should fulfill the requirement)

### Solution
- Add interfaces in the existing flight task components that are in use for Position mode to slow down
- Add a new mode "Position Slow"
- Add a new message for velocity limits which is supposed to be filled with the contents of the MAVLink message from https://github.com/mavlink/mavlink/pull/2015
- Implement a new `FlightTaskManualAccelerationSlow` that orchestrates the inputs

### Changelog Entry
For release notes:
```
Feature: Add new multicopter slowed-down Position mode
Documentation: Missing so far. It needs a new mode page best situated under the Position mode.
```

### Test coverage
The full solution was tested in simulation and on a real quadrotor with different settings and all the different inputs and combinations. It will soon be rolled out to production.

### Context
- I realized in https://github.com/PX4/PX4-Autopilot/pull/22084 that there are probably conflicts with https://github.com/PX4/PX4-Autopilot/pull/20707
   I'm happy to rebase and resolve.
- With https://github.com/mavlink/mavlink/pull/2031 we can also bring in auxiliary remote inputs through MAVLink. I'm adding support for this separately from this pr.

bresch commented on src/modules/flight_mode_manager/tasks/Utility/StickAccelerationXY.cpp:
```diff
@@ -157,7 +159,13 @@ Vector2f StickAccelerationXY::calculateDrag(Vector2f drag_coefficient, const flo
 	const Vector2f velocity_with_sqrt_boost = vel_sp.unit_or_zero() * math::sqrt_linear(vel_sp.norm());
-	return drag_coefficient.emult(velocity_with_sqrt_boost);
+
+	// only apply the drag increase below 1m/s when actually brakeing such that speeds below 1m/s
```
```suggestion
	// only apply the drag increase below 1m/s when actually braking such that speeds below 1m/s
```

bresch replied:
Did you get complaints that the braking was too hard? This will change the braking behavior for everyone, right?

MaEtUgR replied:
Thanks for checking 🙏  I'm fixing the typo.
No the goal of this is to not change the braking behavior but to not map to a different speed if you want to fly below 1m/s in the first place (without braking).

dagar commented:
The branch has both `msg/VelocityLimits.msg` and `msg/velocity_limits.msg`.

MaEtUgR commented:
> The branch has both msg/VelocityLimits.msg and msg/velocity_limits.msg.

Fixed.

I rebased on top of the mode changes.  @tstastny It should be ready now.

MaEtUgR commented:
I had to add ifdef guards for processing the MAVLink message defined in `development.xml` for now in 737317c31b802bdc11847ba7d1601062ec2e4c3c because some targets use `common.xml`. I found that confusing, it also means some targets with the same PX4 version speak some different dialect and it's not easy to find where you could change that for your use case/target. I suggest we quickly speak about that in tomorrow's maintainer call.

MaEtUgR commented:
I tested again, works like expected but now it's over flash 😬 
![image](https://github.com/PX4/PX4-Autopilot/assets/4668506/1b1965c0-e3fe-4404-be2a-d4f02de71a2b)

MaEtUgR commented:
I rebased on main and swapped the mode ID(s) to free ones in the existing range, thanks @bkueng for the idea!

bresch reviewed (approved):
I was wondering if it really makes sense to have it as a separate flight mode rather than just adding constraints in the existing position mode, but why not. I guess it makes it easier to configure (e.g.: existing QGC tab for switches and flight mode name displayed).

MaEtUgR commented:
I added documentation: https://github.com/PX4/PX4-user_guide/pull/2972 😅
